src/Utilities/Temporal.Async/public/TryGetResult.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+namespace Temporal.Async
+{
+    // @ToDo: If we want to support returning these from workflow APIs, we must either implement IDataValue or find some other solution.
+    public sealed class TryGetResult<T>
+    {
+        private readonly bool _isSuccess;
```

Comment:
What minimum version of C# are we allowed to target? Can't you just `public bool IsSuccess { get; init; }`? (I see a lot of C# features, e.g. `var`, properties, etc that are not leveraged making the code quite wordy)